lib/commonio.c

```diff
@@ -965,7 +970,9 @@ int commonio_close (struct commonio_db *db)
 		sb.st_gid = db->st_gid;
 	}
 
-	snprintf (buf, sizeof buf, "%s+", db->filename);
+	r = snprintf (buf, sizeof buf, "%s+", db->filename);
+	if (r < 0 || (size_t)r >= sizeof buf)
```

Comment:
This has me concerned: if line 925 found db->fp != NULL, and so we closed it, we have *not* after that set db->fp = NULL.  So here, if we find this snprintf to have failed, we'll goto fail, and end up in paths that will try to close db->fp again.

I think the simplest and safest way to handle this is to set db->fp = NULL at the bottom of the "if (NULL == db->fp)" block above.

Reply:
Really the 'db->fp = NULL' above there under 'if (errors != 0)' should just be pulled out of that if block, so it just becomes

```
#ifdef WITH_SELINUX
		if (reset_selinux_file_context () != 0) {
			errors++;
		}
#endif
		db->fp = NULL;
		if (errors != 0)
			goto fail;
	} else {

```